eras/shelley/impl/cardano-ledger-shelley.cabal

```
@@ -133,6 +133,53 @@ library
   if flag(asserts)
     ghc-options: -fno-ignore-asserts
 
+library cddl
+  exposed-modules:
+    Cardano.Ledger.Shelley.CDDL
+
+  visibility: public
+  hs-source-dirs: cddl
+  default-language: Haskell2010
+  ghc-options:
+    -Wall
+    -Wcompat
+    -Wincomplete-record-updates
+    -Wincomplete-uni-patterns
+    -Wredundant-constraints
+    -Wpartial-fields
+    -Wunused-packages
+
+  build-depends:
+    base,
+    cardano-ledger-core,
+    cardano-ledger-core:cddl,
+    cardano-ledger-shelley,
+    cuddle >=0.4,
+    heredoc,
+
+executable generate-cddl
+  main-is: Main.hs
+  hs-source-dirs: generate-cddl
```

I think it will be ok to keep it in the same folder as huddle definition. There is no need to clutter the top level project folder
```suggestion
  hs-source-dirs: cddl
```

---

I made these as `cddl/lib` and `cddl/exe` respectively, since having both the files under a single directory caused some `"home-module"` issue.

---

I hope I did the correct thing, that isn't too uncommon.

---

Yeah, that's totally fine. Later we'll be able to move the actual generated cddl files into `cddl/data`. That step will be as the very final step after we fully switched to the type class approach. See my comment: https://github.com/IntersectMBO/cardano-ledger/issues/5194#issuecomment-3554477038